client/js/settings/index/email-settings/index.js

```diff
@@ -16,23 +15,25 @@ const view = require('./view')
 function init() {
   var state = { submit$: flyd.stream() }
 
-  // formSerialize will set checked boxes to "on" and unchecked boxes to "". We want it to be true/false instead
-  const formObj$ = R.compose(
-    flyd.map(obj => R.map(val => val === 'on' ? true : false, obj))
-  , flyd.map(ev => serializeForm(ev.currentTarget, {hash: true, empty: true}))
-  )(state.submit$)
+  const formObj$ = state.submit$.map((ev) => {
+    const formData = serializeForm(ev.currentTarget, { hash: true, empty: true });
+    return Object.keys(formData).reduce((acc, key) => {
+      acc[key] = formData[key] === 'on' ? true : false;
+      return acc;
+    }, {});
+  });
```

Comment:
You're sure this works? This is one of those areas where the original code is just so confusing as to what is intended.

Reply:
Hmmm, I tested it and I'm reasonably confident this works. I guess let's go ahead.